ibm/service/vpc/data_source_ibm_is_instance.go

```diff
@@ -881,6 +913,26 @@ func instanceGetByName(d *schema.ResourceData, meta interface{}, name string) er
 		}
 		d.Set(isInstanceInitKeys, initKeyList)
 	}
+	//set the lifecycle status, reasons
+	if instance.LifecycleState != nil {
+		d.Set(isInstanceLifecycleState, *instance.LifecycleState)
+	}
+	if instance.LifecycleReasons != nil {
+		lifecycleReasonsList := make([]map[string]interface{}, 0)
```

Comment:
can we have seperate function for this instead of duplicating code

Reply:
created a data source flatten method.